gemrb/core/GUI/Tooltip.cpp

```diff
@@ -0,0 +1,165 @@
+/* GemRB - Infinity Engine Emulator
+ * Copyright (C) 2016 The GemRB Project
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "Tooltip.h"
+
+#include "Interface.h"
+#include "GUI/TextSystem/Font.h"
+
+namespace GemRB {
+
+TooltipBackground::TooltipBackground(Holder<Sprite2D> bg, Holder<Sprite2D> left, Holder<Sprite2D> right)
+: background(bg), leftbg(left), rightbg(right)
+{
+	assert(background);
+	assert((leftbg && rightbg) || (!leftbg && !rightbg));
+
+	margin = 5;
+	// no animation by default
+	animationSpeed = 0;
+
+	Reset();
+}
+
+TooltipBackground::TooltipBackground(const TooltipBackground& bg)
+{
+	animationSpeed = bg.animationSpeed;
+	margin = bg.margin;
+
+	background = bg.background;
```

Comment:
Codacy found an issue: [Variable 'background' is assigned in constructor body. Consider performing initialization in initialization list.](https://app.codacy.com/manual/gemrb/gemrb/pullRequest?prid=6310305)